src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -2437,7 +2445,7 @@ protected override void AfterLeftChildHasBeenVisited(BoundBinaryOperator binary)
 
                 VisitRvalue(binary.Right);
                 Debug.Assert(!IsConditionalState);
-                Debug.Assert(this.State.Reachable);
+                //Debug.Assert(this.State.Reachable);
```

Comment:
PROTOTYPE?

Reply:
The assert is too aggressive since it fails with invalid code such as `s + throw new Exception()`. I'll remove the commented out assert.